File: Modules/ServerTypes/PayrollSettings/PayrollDeductionsForm.ts
import { DecimalEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PayrollDeductionsForm {
    Amount: DecimalEditor;
    DeductionCode: StringEditor;
    Description: StringEditor;
}

export class PayrollDeductionsForm extends PrefixedContext {
    static formKey = 'PayrollSettings.PayrollDeductions';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PayrollDeductionsForm.init)  {
            PayrollDeductionsForm.init = true;

            var w0 = DecimalEditor;
            var w1 = StringEditor;

            initFormType(PayrollDeductionsForm, [
                'Amount', w0,
                'DeductionCode', w1,
                'Description', w1
            ]);
        }
    }
}

File: Modules/HumanResource/PayrollSettings/PayrollDeductions/PayrollDeductionsForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.PayrollDeductions")]
[BasedOnRow(typeof(PayrollDeductionsRow), CheckNames = true)]
public class PayrollDeductionsForm
{
    [HalfWidth]
    public double Amount { get; set; }

    [HalfWidth]
    public string DeductionCode { get; set; }

    public string Description { get; set; }

}